python/src/runner.py

```diff
@@ -261,6 +261,7 @@ def __exit__(self, exc_type, value, traceback):
 
 def runCode(fileToRun, globals, args, *, useUntypy=True):
     localDir = os.path.dirname(fileToRun)
+    fileToRun = os.path.realpath(fileToRun) # needed for setting __file__
```

Comment:
Why is is necessary to have to absolute path here? This adds a lot of noise to error messages.

I improved on your commit here:

07a54f46b206bed7282582bcc93121c75f102e4e
7831047dd34aa0cdfd8ea57e273e8ea31ef688a9

and all tests are running.

Reply:
der gehört auch noch dazu:

9c3a3ac5b09825effe02988f1902821637e04126

Reply:
`fileToRun = os.path.realpath(fileToRun)`
If the current working directory is changed the path of `__file__` would be become invalid.

Reply:
Not quite because we are simply using `fileToRun` for loading the code and no `os.chdir` happens after loading the code. So the path must be correct at this point.

Reply:
`os.chdir` could be called by the user. As far as i know the file is read in again by the inspect-module when the first type error occurs.

Reply:
that's a good point although I think this situation occurs rarely. I've created an issue: https://github.com/skogsbaer/write-your-python-program/issues/26